portalnet/src/overlay/protocol.rs

```diff
@@ -531,6 +531,32 @@ where
             destination: enr.clone(),
         };
 
+        // Validate that the content keys are available in the local store, before sending the
+        // offer
+        for content_key in content_keys.into_iter() {
+            let content_key = TContentKey::try_from(content_key.clone()).map_err(|err| {
+                OverlayRequestError::ContentNotFound {
+                    message: format!(
+                        "Error decoding content key for content key: {content_key:02X?} - {err}"
+                    ),
+                    utp: false,
+                    trace: None,
+                }
+            })?;
+            match self.store.read().get(&content_key) {
```

Comment:
Question more for my understanding: considering that now we are getting the value from the DB before the OFFER request anyway, why not use `Request::PopulatedOffer` instead?

And if we switch to `Request::PopulatedOffer`, do we ever use `Request::Offer`? And if not, can we remove it?

Reply:
I wasn't talking about JSON rpc endpoint, as much as I was talking about `Request` object and how we handle it internally.

For example, the json rpc handler can retrieve the content value from the local storage, and call `network.overlay.send_populated_offer(..)` instead. In that case, I believe we don't need `Request::Offer` type at all (but still support json offer endpoint).

If you are still not sure what I mean, we can have a look at the code together. It's very possible that I'm not understanding how some things work.

Reply:
Ahh, ok yeah that's interesting. Off the top of my head that seems feasible, but will probably need to do a little tinkering to confirm and make sure there aren't any edge cases that wouldn't be supported. Not super high-priority imo but I put it on my flamingo list of todos